rclcpp_lifecycle/src/lifecycle_talker.cpp

```diff
@@ -0,0 +1,87 @@
+// Copyright 2016 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <iostream>
+#include <memory>
+
+#include "rclcpp/rclcpp.hpp"
+#include "rclcpp/publisher.hpp"
+
+#include "rclcpp_lifecycle/lifecycle_node.hpp"
+#include "rclcpp_lifecycle/lifecycle_publisher.hpp"
+
+#include "std_msgs/msg/string.hpp"
+
+int main(int argc, char * argv[])
+{
+  rclcpp::init(argc, argv);
+
+  std::shared_ptr<rclcpp::node::LifecycleNode> lc_node = std::make_shared<rclcpp::node::LifecycleNode>("lc_talker");
+ 
+  rmw_qos_profile_t custom_qos_profile = rmw_qos_profile_default;
+  custom_qos_profile.depth = 7;
+
+  std::shared_ptr<rclcpp::publisher::LifecyclePublisher<std_msgs::msg::String>> chatter_pub =
+    lc_node->create_publisher<std_msgs::msg::String>("chatter", custom_qos_profile);
+
+  lc_node->print_state_machine();
+
+  if (!lc_node->on_configure())
```

Comment:
The external API of a managed lifecycle node for controlling transitions between these states is not the callbacks. There should be an API containing the 7 transitions mentioned in the design document (`create`, `configure`, etc.) When, for example, `configure()` is called on the node, it should execute the `on_configure()` callback and shift its state to the appropriate successor state based on the result.

See also my comment in the managed lifecycle node implementation.